src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -474,6 +473,16 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator, IRBlobSe
         private async Task RunWorker(CancellationToken ct) {
             TaskUtilities.AssertIsOnBackgroundThread();
 
+            // Spin until the worker task is registered.
+            while (_runTask == null) {
+                await Task.Yield();
```

Comment:
I'm starting to think that the entire existing arrangement is a hack, and we should just use a semaphore here to ensure that there's only one worker task... but this at least fixes the issue that was there. FWIW, this isn't directly related to other changes, it's a race that we had before that I've ran into while doing multiple verification test runs.

Can you file a bug detailing what you think is the best approach to rewrite this part? And we can do a separate PR for that then.

---
In reply to: [74950109](https://github.com/Microsoft/RTVS/pull/2173#discussion_r74950109) [](ancestors = 74950109)